integration-test/src/main/java/com/sequenceiq/it/cloudbreak/CredentialModifyClusterTests.java

```diff
@@ -0,0 +1,259 @@
+import org.testng.Assert;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import javax.ws.rs.ForbiddenException;
+import javax.ws.rs.WebApplicationException;
+import javax.ws.rs.core.Response;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class CredentialModifyClusterTests extends CloudbreakClusterTestConfiguration {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(CredentialModifyClusterTests.class);
+
+    @Test(dataProvider = "providernameblueprintimage", priority = 10)
+    public void testCreateNewRegularCluster(CloudProvider cloudProvider, String clusterName, String credentialName, String blueprintName, String imageId)
+            throws Exception {
+        given(CloudbreakClient.created());
+        given(cloudProvider.aValidCredential()
+                .withName(credentialName), "a credential is created.");
+        given(Cluster.request()
+                .withAmbariRequest(cloudProvider.ambariRequestWithBlueprintName(blueprintName)),
+                "a cluster request");
+        given(ImageSettings.request()
+                .withImageCatalog("")
+                .withImageId(imageId));
+        given(cloudProvider.aValidStackRequest()
+                .withName(clusterName), "a stack request");
+        when(Stack.post(), "post the stack request");
+        then(Stack.waitAndCheckClusterAndStackAvailabilityStatus(),
+                "wait and check availability");
+        then(Stack.checkClusterHasAmbariRunning(
+                getTestParameter().get(CloudProviderHelper.DEFAULT_AMBARI_PORT),
+                getTestParameter().get(CloudProviderHelper.DEFAULT_AMBARI_USER),
+                getTestParameter().get(CloudProviderHelper.DEFAULT_AMBARI_PASSWORD)),
+                "check ambari is running and components available");
+    }
+
+    @Test(dataProvider = "providername", priority = 20)
+    public void testModifyAWSCredential(CloudProvider cloudProvider, String clusterName, String credentialName) throws Exception {
+        AwsCloudProvider provider = new AwsCloudProvider(getTestParameter());
+
+        LOGGER.debug("Credential Name is ::: {}", credentialName);
+        given(CloudbreakClient.created());
+        given(provider.aValidCredential()
+                .withName(credentialName), "a credential is created.");
+        given(Credential.request()
+                .withName(credentialName)
+                .withParameters(provider.awsCredentialDetailsKey())
+                .withCloudPlatform(provider.getPlatform()), " credential Key Based modification is requested.");
+        when(Credential.put(), " credential has been modified.");
+        then(Credential.assertThis(
+                (credential, t) -> {
+                    Map<String, Object> parameterMappings = credential.getResponse().getParameters();
```

Comment:
Since we're not using this variable anymore just for iterating over its entry set, it is not necessary to assign it separately. Easier to pass the _credential.getResponse().getParameters()_ directly to the foreach.